src/backend/distributed/commands/function.c

```diff
@@ -615,3 +632,492 @@ TriggerSyncMetadataToPrimaryNodes(void)
 		TriggerMetadataSync(MyDatabaseId);
 	}
 }
+
+
+/*
+ * ShouldPropagateAlterFunction returns, based on the address of a function, if alter
+ * statements targeting the function should be propagated.
+ */
+static bool
+ShouldPropagateAlterFunction(const ObjectAddress *address)
+{
+	if (creating_extension)
+	{
+		/*
+		 * extensions should be created separately on the workers, functions cascading
+		 * from an extension should therefore not be propagated.
+		 */
+		return false;
+	}
+
+	if (!EnableDependencyCreation)
+	{
+		/*
+		 * we are configured to disable object propagation, should not propagate anything
+		 */
+		return false;
+	}
+
+	if (!IsObjectDistributed(address))
+	{
+		/* do not propagate alter function for non-distributed functions */
+		return false;
+	}
+
+	return true;
+}
+
+
+/*
+ * PlanAlterFunctionStmt is invoked for alter function statements with actions. Here we
+ * plan the jobs to be executed on the workers for functions that have been distributed in
+ * the cluster.
+ */
+List *
+PlanAlterFunctionStmt(AlterFunctionStmt *stmt, const char *queryString)
+{
+	const char *sql = NULL;
+	const ObjectAddress *address = NULL;
+	List *commands = NIL;
+
+#if PG_VERSION_NUM > 110000
+	AssertIsFunctionOrProcedure(stmt->objtype);
+#endif
+
+	address = GetObjectAddressFromParseTree((Node *) stmt, false);
+	if (!ShouldPropagateAlterFunction(address))
+	{
+		return NIL;
+	}
+
+	EnsureCoordinator();
+
+	ErrorIfUnsupportedAlterFunctionStmt(stmt);
+
+	/* reconstruct alter statement in a portable fashion */
+	QualifyTreeNode((Node *) stmt);
+	sql = DeparseTreeNode((Node *) stmt);
+
+	/*
+	 * all functions that are distributed will need their alter statements propagated
```

Comment:
Super minor: Maybe move `EnsureSequentialModeForFunctionDDL ` above `/* reconstruct alter statement in a portable fashion */` block, the flow would look nicer. (In general, this applies to other functions as well, but feel free to ignore.)